ocs_ci/ocs/registry.py

```diff
@@ -343,3 +346,66 @@ def check_image_exists_in_registry(image_url):
         return_value = True
         logger.info("Image exists in Registry")
     return return_value
+
+
+def image_pull_and_push(
+    project_name, template, image='', pattern='', wait=True
+):
+    """
+    Pull and push images running oc new-app command
+
+    Args:
+        project_name (str): Name of project
+        template (str): Name of the template of the image
+        image (str): Name of the image with tag
+        pattern (str): name of the build with given pattern
+        wait (bool): If true waits till the image pull and push completes.
+
+    """
+    ocp_obj = ocp.OCP()
+    ocp_obj.new_project(project_name=project_name)
+    if config.DEPLOYMENT.get('disconnected'):
+        mirror_image(image=image)
+    else:
+        cmd = f"new-app --template={template} -n {project_name}"
+        ocp_obj.exec_oc_cmd(command=cmd, out_yaml_format=False)
+
+        # Validate it completed
+        if wait:
+            wait_time = 300
+            logger.info(f"Wait for {wait_time} seconds for build to come up")
+            time.sleep(300)
+            build_list = get_build_name_by_pattern(pattern=pattern, namespace=project_name)
+            assert build_list is not None, "Build is not created"
```

Comment:
Do you think we should throw exception here instead of assert?

Reply:
done